### PR TITLE
Messages table

### PR DESCRIPTION
**Pull request by hishamshirbini**
tables and header

### CODE REVIEW

**Comment by ziad007** [comment]
@hishamshirbini Can we grouped services in folder Services and mock file under Mock Folder

File: src/app/reports/reports.component.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  add(message: string): void {
+    message = message.trim();
+    
+    var date = formatDate(new Date(), 'MMM dd, yyyy', 'en');
+    var userName = 'Jhon Doe';
+    var status = 'new-message'
+    if (!message) { return; }
+    this.reportService.addReport({ userName , date , message ,status} as Report)
```
Comment (ziad007) [inline src/app/reports/reports.component.ts]: ({userName , date, message, status}

File: src/app/investors/investors.component.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  </tr>
+
+  <tr *ngFor="let investor of investors">
+    <td class="table-investor-name">
+        {{investor.name}}
+      </td>
+    <td></td>
+    <td class="table-date">
+      <i class="fa fa-user user-icon-investor" ></i>
+
```
[the remark this answers is not in the document]
Comment (hishamshirbini) [reply src/app/investors/investors.component.html]: <td class="table-date">
      <i class="fa fa-user user-icon-investor" ></i>
  </tr>
</table>

File: src/app/investor.service.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  constructor(
+    private http: HttpClient,
+    private messageService: MessageService) { }
+
+  /** GET investors from the server */
+  getInvestors (): Observable<Investor[]> {
+    return this.http.get<Investor[]>(this.investorsUrl)
+      .pipe(
+        tap(_ => this.log('fetched investors')),
+        catchError(this.handleError<Investor[]>('getInvestores', []))
```
[the remark this answers is not in the document]
Comment (hishamshirbini) [reply src/app/investor.service.ts]: .pipe(
        tap(_ => this.log('fetched investors')),
        catchError(this.handleError<Investor[]>('getInvestors', []))
      );
  }

**Pull request merged**